lib/system/shared/mesh-client.js

```diff
@@ -221,7 +221,17 @@
 
     var client = {
       _mesh: {},
-      log: opts.log
+      log: opts.log,
+      updateSession:function(session){
+        var _this = this;
+        _this.session = session;
+        if (_this._mesh && _this._mesh.endpoints){
+          Object.keys(_this._mesh.endpoints).forEach(function(endpointName){
+              var endpoint = _this._mesh.endpoints[endpointName];
+              endpoint.messenger.session.id = session.id;
```

Comment:
1. the session id is used by the client for the request and response paths - and is not connected to the endpoint messenger via a reference - so it gets set initially and has to be reset, per endpoint when it changes.

2. good question!! I just realised that older versions of happn were just working, because the client messenger is listening on its old session id and the server is still emitting to everyone on the session path. I think this will be backward compatible, because the old clients will still be pushing on the old id, and will receive on the old one, but I will create a test to prove that clients both new and updated are able to do exchange methods before we do a release.